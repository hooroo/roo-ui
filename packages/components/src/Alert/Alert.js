import React from 'react';
import PropTypes from 'prop-types';
import cleanElement from 'clean-element';

import { Container, NakedButton, Text, Flex, Box, Icon } from '..';

const CleanDiv = cleanElement('div');

const Base = ({
  children,
  icon,
  onClose,
  ...props
}) => (
  <CleanDiv {...props}>
    <Container>
      <Flex py={4}>
        {icon && (
          <Box pl={1} pr={3}>
            <Icon {...icon} />
          </Box>
        )}

        <Text mb={0} textAlign="left">
          {children}
        </Text>

        {onClose && (
          <Box ml="auto" pr={1} pl={3}>
            <NakedButton onClick={onClose}>
              <Icon name="close" />
            </NakedButton>
          </Box>
        )}
      </Flex>
    </Container>
  </CleanDiv>
);

Base.propTypes = {
  children: PropTypes.node.isRequired,
  icon: PropTypes.shape(Icon.propTypes),
  onClose: PropTypes.func,
};

Base.defaultProps = {
  icon: null,
  onClose: null,
};

const Alert = Box.withComponent(Base);
Alert.defaultProps = {
  ...Box.defaultProps,
  bg: 'grey.3',
  mb: 3,
};

Alert.info = Alert.extend``;
Alert.info.displayName = 'Alert';
Alert.info.defaultProps = {
  ...Alert.defaultProps,
  bg: 'ui.infoBackground',
  icon: {
    name: 'info',
    color: 'grey.0',
  },
};

Alert.success = Alert.extend``;
Alert.success.displayName = 'Alert';
Alert.success.defaultProps = {
  ...Alert.defaultProps,
  bg: 'ui.successBackground',
  icon: {
    name: 'checkCircle',
    color: 'ui.success',
  },
};

Alert.error = Alert.extend``;
Alert.error.displayName = 'Alert';
Alert.error.defaultProps = {
  ...Alert.defaultProps,
  bg: 'ui.errorBackground',
  icon: {
    name: 'warning',
    color: 'ui.error',
  },
};

export default Alert;
