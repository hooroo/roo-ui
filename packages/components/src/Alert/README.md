# Alert

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Alert } from '@roo-ui/components';

export default (
  <Alert onClose={closeAlert} icon={{ name: 'hotel' }}>
    Hello world
  </Alert>
);
```

## Properties

| Name        | Description                                                          | Type     | Default   | Required? |
|-------------|----------------------------------------------------------------------|----------|-----------|-----------|
| `onClose`   | called when close icon is clicked                                    | `func`   | -         | -         |
| `contained` | wrap the contents of the <Alert /> in a <Container />                | `bool`   | false     | -         |
| `icon`      | render an icon                                                       | `shape`  | -         | -         |
| `variant`      | `alertStyle` from theme. One of `default`, `info`, `success` or `error` | `string` | 'default' | -         |

Refer to [<Icon />](../Icon) for available `icon` properties.
