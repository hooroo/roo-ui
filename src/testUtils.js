import React from 'react';
import { mount, shallow } from 'enzyme';

export const mountWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return mount(themedTree);
};

export const shallowWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return shallow(themedTree);
};
