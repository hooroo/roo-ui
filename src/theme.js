import { rem } from 'polished';
import range from 'lodash/range';

import * as icons from './defaultIcons';

const namedColors = {
  white: '#FFFFFF',
  brightSun: '#FBCB3B',
  red: '#E00E00',
  maroon: '#870000',
  blue: '#8DE2E0',
  lightBlue: '#E7F7F7',
  orange: '#ED710B',
  lightOrange: '#FCEBCD',
  green: '#35A509',
  lightGreen: '#DEF2DE',
};

const brand = {
  primary: '#E40000',
  secondary: '#8DE2E0',
  tertiary: '#D20000',
  quaternary: '#F9F3E9',
};

const greys = {
  charcoal: '#323232',
  steel: '#666666',
  alto: '#DADADA',
  porcelain: '#F4F5F6',
  dusty: '#999999',
  mineShaft: '#222222',
};

const ui = {
  error: namedColors.orange,
  errorBackground: namedColors.lightOrange,
  info: namedColors.blue,
  infoBackground: namedColors.lightBlue,
  link: namedColors.red,
  linkHover: namedColors.maroon,
  success: namedColors.green,
  successBackground: namedColors.lightGreen,
};

const colors = {
  brand,
  greys,
  ui,
  ...namedColors,
};

const alertStyles = {
  default: { bg: 'greys.porcelain' },
  error: {
    bg: 'ui.errorBackground',
    icon: {
      name: 'warning',
      color: 'ui.error',
    },
  },
  info: {
    bg: 'ui.infoBackground',
    icon: {
      name: 'info',
      color: 'greys.charcoal',
    },
  },
  success: {
    bg: 'ui.successBackground',
    icon: {
      name: 'checkCircle',
      color: 'ui.success',
    },
  },
};

const borders = [0, '1px solid', '2px solid', '3px solid'];

const breakpoints = ['40rem', '52rem', '64rem'];

const buttons = {
  default: {
    color: 'white',
    backgroundColor: 'greys.charcoal',
  },
  primary: {
    color: 'white',
    backgroundColor: 'brand.primary',
  },
};

const fontFamily = 'Ciutadella, sans-serif';

const fontSizes = {
  xs: rem('13px'),
  sm: rem('15px'),
  base: rem('16px'),
  lg: rem('18px'),
  xl: rem('22px'),
  '2xl': rem('26px'),
  '3xl': rem('36px'),
  '4xl': rem('46px'),
};

const fontWeights = {
  normal: 500,
  bold: 600,
};

const letterSpacings = {
  normal: '0',
  wide: '0.125em',
};

const lineHeights = {
  tight: 1,
  normal: 1.35,
  loose: 1.5,
};

const maxWidths = {
  default: rem('1280px'),
};

const mediaQueries = breakpoints.map(
  n => `@media screen and (min-width: ${n})`,
);

const opacity = {
  disabled: 0.7,
};

const radii = {
  none: 0,
  default: '4px',
  rounded: '1000px',
};

const shadows = {
  none: 0,
  default: '0 1px 1px 0 rgba(0, 0, 0, 0.1)',
  heavy: '0 2px 2px 0 rgba(0, 0, 0, 0.1)',
  focus: `0 0 2px 2px ${colors.greys.dusty};`,
};

const space = range(0, 129, 4).map(i => rem(`${i}px`));

const textStyles = {
  caps: {
    textTransform: 'uppercase',
    fontWeight: fontWeights.bold,
  },

  h1: {
    fontSize: fontSizes['3xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h2: {
    fontSize: fontSizes['2xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h3: {
    fontSize: fontSizes.xl,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h4: {
    fontSize: fontSizes.lg,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h5: {
    fontSize: fontSizes.base,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h6: {
    fontSize: fontSizes.sm,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  label: {
    fontSize: fontSizes.sm,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.bold,
    color: colors.greys.charcoal,
  },
  text: {
    fontSize: fontSizes.base,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.normal,
  },
};

const transitions = {
  default: '200ms ease-in',
};

export default {
  alertStyles,
  borders,
  breakpoints,
  buttons,
  colors,
  fontFamily,
  fontSizes,
  fontWeights,
  icons,
  letterSpacings,
  lineHeights,
  maxWidths,
  mediaQueries,
  opacity,
  radii,
  shadows,
  space,
  textStyles,
  transitions,
};
