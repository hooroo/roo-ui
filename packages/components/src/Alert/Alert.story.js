import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';
import { boolean } from '@storybook/addon-knobs/react';

import Alert from './Alert';
import README from './README.md';

storiesOf('Components|Alert', module)
  .addDecorator(withDocs(README))
  .add('<Alert />', () => (
    <Alert contained={boolean('Contained', false)} icon={{ name: 'hotel' }}>
      <strong>Hello world</strong><br />
      Lorem ipsum dolor sit amet
    </Alert>
  ))
  .add('<Alert.info />', () => (
    <Alert.info onClose={() => {}}>
      <strong>Info</strong><br />
      Lorem ipsum dolor sit amet
    </Alert.info>
  ))
  .add('<Alert.success />', () => (
    <Alert.success onClose={() => {}}>
      <strong>Success</strong><br />
      Lorem ipsum dolor sit amet
    </Alert.success>
  ))
  .add('<Alert.error />', () => (
    <Alert.error onClose={() => {}}>
      <strong>Error</strong><br />
      Lorem ipsum dolor sit amet
    </Alert.error>
  ))
  .add('<Alert variant="info" />', () => (
    <Alert variant="info" onClose={() => {}}>
      <strong>Info</strong><br />
      Lorem ipsum dolor sit amet
    </Alert>
  ))
  .add('<Alert variant="success" />', () => (
    <Alert variant="success" onClose={() => {}}>
      <strong>Success</strong><br />
      Lorem ipsum dolor sit amet
    </Alert>
  ))
  .add('<Alert variant="error" />', () => (
    <Alert variant="error" onClose={() => {}}>
      <strong>Error</strong><br />
      Lorem ipsum dolor sit amet
    </Alert>
  ));

