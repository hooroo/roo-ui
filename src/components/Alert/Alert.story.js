import React from 'react';
import { withDocs } from 'storybook-readme';
import { boolean } from '@storybook/addon-knobs';

import Alert from './Alert';
import README from './README.md';

export default {
  title: 'Components/Alert',
  decorators: [withDocs(README)],
};

export const _Alert = () => (
  <Alert contained={boolean('Contained', false)} icon={{ name: 'hotel' }}>
    <strong>Hello world</strong>
    <br />
    Lorem ipsum dolor sit amet
  </Alert>
);

_Alert.story = {
  name: '<Alert />',
};

export const AlertInfo = () => (
  <Alert.info onClose={() => {}}>
    <strong>Info</strong>
    <br />
    Lorem ipsum dolor sit amet
  </Alert.info>
);

AlertInfo.story = {
  name: '<Alert.info />',
};

export const AlertSuccess = () => (
  <Alert.success onClose={() => {}}>
    <strong>Success</strong>
    <br />
    Lorem ipsum dolor sit amet
  </Alert.success>
);

AlertSuccess.story = {
  name: '<Alert.success />',
};

export const AlertError = () => (
  <Alert.error onClose={() => {}}>
    <strong>Error</strong>
    <br />
    Lorem ipsum dolor sit amet
  </Alert.error>
);

AlertError.story = {
  name: '<Alert.error />',
};

export const AlertVariantInfo = () => (
  <Alert variant="info" onClose={() => {}}>
    <strong>Info</strong>
    <br />
    Lorem ipsum dolor sit amet
  </Alert>
);

AlertVariantInfo.story = {
  name: '<Alert variant="info" />',
};

export const AlertVariantSuccess = () => (
  <Alert variant="success" onClose={() => {}}>
    <strong>Success</strong>
    <br />
    Lorem ipsum dolor sit amet
  </Alert>
);

AlertVariantSuccess.story = {
  name: '<Alert variant="success" />',
};

export const AlertVariantError = () => (
  <Alert variant="error" onClose={() => {}}>
    <strong>Error</strong>
    <br />
    Lorem ipsum dolor sit amet
  </Alert>
);

AlertVariantError.story = {
  name: '<Alert variant="error" />',
};
