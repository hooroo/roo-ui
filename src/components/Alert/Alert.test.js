import React from 'react';
import theme from 'theme';
import { mountWithTheme } from 'testUtils';
import { axe } from 'jest-axe';

import Alert from '.';

describe('<Alert />', () => {
  let wrapper;

  const render = tree => mountWithTheme(tree, theme);

  beforeEach(() => {
    wrapper = render(<Alert>Hello world</Alert>);
  });

  it('has expected displayName', () => {
    expect(Alert.displayName).toBe('Alert');
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  describe('contained', () => {
    beforeEach(() => {
      wrapper = render(<Alert contained>Hello world</Alert>);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Alert.success />', () => {
    beforeEach(() => {
      wrapper = render(<Alert.success>Hello world</Alert.success>);
    });

    it('has expected displayName', () => {
      expect(Alert.success.displayName).toBe('Alert.success');
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Alert variant="success" />', () => {
    beforeEach(() => {
      wrapper = render(<Alert variant="success">Hello world</Alert>);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Alert.error />', () => {
    beforeEach(() => {
      wrapper = render(<Alert.error>Hello world</Alert.error>);
    });

    it('has expected displayName', () => {
      expect(Alert.error.displayName).toBe('Alert.error');
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Alert variant="error" />', () => {
    beforeEach(() => {
      wrapper = render(<Alert variant="error">Hello world</Alert>);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Alert.info />', () => {
    beforeEach(() => {
      wrapper = render(<Alert.info>Hello world</Alert.info>);
    });

    it('has expected displayName', () => {
      expect(Alert.info.displayName).toBe('Alert.info');
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Alert variant="info" />', () => {
    beforeEach(() => {
      wrapper = render(<Alert variant="info">Hello world</Alert>);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('on close', () => {
    let onClose;

    beforeEach(() => {
      onClose = jest.fn();
      wrapper = render(<Alert onClose={onClose}>Hello world</Alert>);
      wrapper.find('NakedButton').simulate('click');
    });

    it('calls #onClose', () => {
      expect(onClose).toHaveBeenCalled();
    });
  });
});
