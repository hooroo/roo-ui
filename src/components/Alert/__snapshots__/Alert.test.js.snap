// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> <Alert variant="error" /> renders correctly 1`] = `
.emotion-15 {
  padding-left: 32px;
  padding-right: 32px;
}

.emotion-13 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  margin-bottom: 0;
  text-align: left;
}

.emotion-9 {
  padding-right: 16px;
}

.emotion-17 {
  margin-bottom: 16px;
  background-color: ui.errorBackground;
}

.emotion-3 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.error;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.error;
}

<Alert
  mb={3}
  variant="error"
>
  <Component
    mb={3}
    variant="error"
  >
    <Box
      bg="ui.errorBackground"
      mb={3}
    >
      <div
        className="emotion-17 emotion-10"
      >
        <Box
          px={4}
        >
          <div
            className="emotion-15 emotion-10"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-13 emotion-14"
              >
                <Box
                  pr={3}
                >
                  <div
                    className="emotion-9 emotion-10"
                  >
                    <Icon
                      color="ui.error"
                      name="warning"
                      size={24}
                      title={null}
                    >
                      <WithTheme(Component)
                        className="emotion-3 emotion-0"
                        color="ui.error"
                        name="warning"
                        size={24}
                        title={null}
                      >
                        <Component
                          className="emotion-3 emotion-0"
                          color="ui.error"
                          name="warning"
                          size={24}
                          title={null}
                        >
                          <StyledSvg
                            className="emotion-3 emotion-0"
                            color="ui.error"
                            fill="currentcolor"
                            height={24}
                            title="warning"
                            viewBox="0 0 24 24"
                            width={24}
                          >
                            <svg
                              className="emotion-0 emotion-1 emotion-2"
                              color="ui.error"
                              fill="currentcolor"
                              height={24}
                              title="warning"
                              viewBox="0 0 24 24"
                              width={24}
                            >
                              <path
                                d={null}
                              />
                            </svg>
                          </StyledSvg>
                        </Component>
                      </WithTheme(Component)>
                    </Icon>
                  </div>
                </Box>
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-11 emotion-12"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Box>
      </div>
    </Box>
  </Component>
</Alert>
`;

exports[`<Alert /> <Alert variant="info" /> renders correctly 1`] = `
.emotion-15 {
  padding-left: 32px;
  padding-right: 32px;
}

.emotion-13 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  margin-bottom: 0;
  text-align: left;
}

.emotion-9 {
  padding-right: 16px;
}

.emotion-17 {
  margin-bottom: 16px;
  background-color: ui.infoBackground;
}

.emotion-3 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: greys.charcoal;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: greys.charcoal;
}

<Alert
  mb={3}
  variant="info"
>
  <Component
    mb={3}
    variant="info"
  >
    <Box
      bg="ui.infoBackground"
      mb={3}
    >
      <div
        className="emotion-17 emotion-10"
      >
        <Box
          px={4}
        >
          <div
            className="emotion-15 emotion-10"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-13 emotion-14"
              >
                <Box
                  pr={3}
                >
                  <div
                    className="emotion-9 emotion-10"
                  >
                    <Icon
                      color="greys.charcoal"
                      name="info"
                      size={24}
                      title={null}
                    >
                      <WithTheme(Component)
                        className="emotion-3 emotion-0"
                        color="greys.charcoal"
                        name="info"
                        size={24}
                        title={null}
                      >
                        <Component
                          className="emotion-3 emotion-0"
                          color="greys.charcoal"
                          name="info"
                          size={24}
                          title={null}
                        >
                          <StyledSvg
                            className="emotion-3 emotion-0"
                            color="greys.charcoal"
                            fill="currentcolor"
                            height={24}
                            title="info"
                            viewBox="0 0 24 24"
                            width={24}
                          >
                            <svg
                              className="emotion-0 emotion-1 emotion-2"
                              color="greys.charcoal"
                              fill="currentcolor"
                              height={24}
                              title="info"
                              viewBox="0 0 24 24"
                              width={24}
                            >
                              <path
                                d={null}
                              />
                            </svg>
                          </StyledSvg>
                        </Component>
                      </WithTheme(Component)>
                    </Icon>
                  </div>
                </Box>
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-11 emotion-12"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Box>
      </div>
    </Box>
  </Component>
</Alert>
`;

exports[`<Alert /> <Alert variant="success" /> renders correctly 1`] = `
.emotion-15 {
  padding-left: 32px;
  padding-right: 32px;
}

.emotion-13 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  margin-bottom: 0;
  text-align: left;
}

.emotion-17 {
  margin-bottom: 16px;
  background-color: ui.successBackground;
}

.emotion-9 {
  padding-right: 16px;
}

.emotion-3 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.success;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.success;
}

<Alert
  mb={3}
  variant="success"
>
  <Component
    mb={3}
    variant="success"
  >
    <Box
      bg="ui.successBackground"
      mb={3}
    >
      <div
        className="emotion-17 emotion-10"
      >
        <Box
          px={4}
        >
          <div
            className="emotion-15 emotion-10"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-13 emotion-14"
              >
                <Box
                  pr={3}
                >
                  <div
                    className="emotion-9 emotion-10"
                  >
                    <Icon
                      color="ui.success"
                      name="checkCircle"
                      size={24}
                      title={null}
                    >
                      <WithTheme(Component)
                        className="emotion-3 emotion-0"
                        color="ui.success"
                        name="checkCircle"
                        size={24}
                        title={null}
                      >
                        <Component
                          className="emotion-3 emotion-0"
                          color="ui.success"
                          name="checkCircle"
                          size={24}
                          title={null}
                        >
                          <StyledSvg
                            className="emotion-3 emotion-0"
                            color="ui.success"
                            fill="currentcolor"
                            height={24}
                            title="checkCircle"
                            viewBox="0 0 24 24"
                            width={24}
                          >
                            <svg
                              className="emotion-0 emotion-1 emotion-2"
                              color="ui.success"
                              fill="currentcolor"
                              height={24}
                              title="checkCircle"
                              viewBox="0 0 24 24"
                              width={24}
                            >
                              <path
                                d={null}
                              />
                            </svg>
                          </StyledSvg>
                        </Component>
                      </WithTheme(Component)>
                    </Icon>
                  </div>
                </Box>
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-11 emotion-12"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Box>
      </div>
    </Box>
  </Component>
</Alert>
`;

exports[`<Alert /> <Alert.error /> renders correctly 1`] = `
.emotion-15 {
  padding-left: 32px;
  padding-right: 32px;
}

.emotion-13 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  margin-bottom: 0;
  text-align: left;
}

.emotion-9 {
  padding-right: 16px;
}

.emotion-17 {
  margin-bottom: 16px;
  background-color: ui.errorBackground;
}

.emotion-3 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.error;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.error;
}

<Alert.error
  mb={3}
  variant="error"
>
  <Component
    mb={3}
    variant="error"
  >
    <Box
      bg="ui.errorBackground"
      mb={3}
    >
      <div
        className="emotion-17 emotion-10"
      >
        <Box
          px={4}
        >
          <div
            className="emotion-15 emotion-10"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-13 emotion-14"
              >
                <Box
                  pr={3}
                >
                  <div
                    className="emotion-9 emotion-10"
                  >
                    <Icon
                      color="ui.error"
                      name="warning"
                      size={24}
                      title={null}
                    >
                      <WithTheme(Component)
                        className="emotion-3 emotion-0"
                        color="ui.error"
                        name="warning"
                        size={24}
                        title={null}
                      >
                        <Component
                          className="emotion-3 emotion-0"
                          color="ui.error"
                          name="warning"
                          size={24}
                          title={null}
                        >
                          <StyledSvg
                            className="emotion-3 emotion-0"
                            color="ui.error"
                            fill="currentcolor"
                            height={24}
                            title="warning"
                            viewBox="0 0 24 24"
                            width={24}
                          >
                            <svg
                              className="emotion-0 emotion-1 emotion-2"
                              color="ui.error"
                              fill="currentcolor"
                              height={24}
                              title="warning"
                              viewBox="0 0 24 24"
                              width={24}
                            >
                              <path
                                d={null}
                              />
                            </svg>
                          </StyledSvg>
                        </Component>
                      </WithTheme(Component)>
                    </Icon>
                  </div>
                </Box>
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-11 emotion-12"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Box>
      </div>
    </Box>
  </Component>
</Alert.error>
`;

exports[`<Alert /> <Alert.info /> renders correctly 1`] = `
.emotion-15 {
  padding-left: 32px;
  padding-right: 32px;
}

.emotion-13 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  margin-bottom: 0;
  text-align: left;
}

.emotion-9 {
  padding-right: 16px;
}

.emotion-17 {
  margin-bottom: 16px;
  background-color: ui.infoBackground;
}

.emotion-3 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: greys.charcoal;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: greys.charcoal;
}

<Alert.info
  mb={3}
  variant="info"
>
  <Component
    mb={3}
    variant="info"
  >
    <Box
      bg="ui.infoBackground"
      mb={3}
    >
      <div
        className="emotion-17 emotion-10"
      >
        <Box
          px={4}
        >
          <div
            className="emotion-15 emotion-10"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-13 emotion-14"
              >
                <Box
                  pr={3}
                >
                  <div
                    className="emotion-9 emotion-10"
                  >
                    <Icon
                      color="greys.charcoal"
                      name="info"
                      size={24}
                      title={null}
                    >
                      <WithTheme(Component)
                        className="emotion-3 emotion-0"
                        color="greys.charcoal"
                        name="info"
                        size={24}
                        title={null}
                      >
                        <Component
                          className="emotion-3 emotion-0"
                          color="greys.charcoal"
                          name="info"
                          size={24}
                          title={null}
                        >
                          <StyledSvg
                            className="emotion-3 emotion-0"
                            color="greys.charcoal"
                            fill="currentcolor"
                            height={24}
                            title="info"
                            viewBox="0 0 24 24"
                            width={24}
                          >
                            <svg
                              className="emotion-0 emotion-1 emotion-2"
                              color="greys.charcoal"
                              fill="currentcolor"
                              height={24}
                              title="info"
                              viewBox="0 0 24 24"
                              width={24}
                            >
                              <path
                                d={null}
                              />
                            </svg>
                          </StyledSvg>
                        </Component>
                      </WithTheme(Component)>
                    </Icon>
                  </div>
                </Box>
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-11 emotion-12"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Box>
      </div>
    </Box>
  </Component>
</Alert.info>
`;

exports[`<Alert /> <Alert.success /> renders correctly 1`] = `
.emotion-15 {
  padding-left: 32px;
  padding-right: 32px;
}

.emotion-13 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  margin-bottom: 0;
  text-align: left;
}

.emotion-17 {
  margin-bottom: 16px;
  background-color: ui.successBackground;
}

.emotion-9 {
  padding-right: 16px;
}

.emotion-3 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.success;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: ui.success;
}

<Alert.success
  mb={3}
  variant="success"
>
  <Component
    mb={3}
    variant="success"
  >
    <Box
      bg="ui.successBackground"
      mb={3}
    >
      <div
        className="emotion-17 emotion-10"
      >
        <Box
          px={4}
        >
          <div
            className="emotion-15 emotion-10"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-13 emotion-14"
              >
                <Box
                  pr={3}
                >
                  <div
                    className="emotion-9 emotion-10"
                  >
                    <Icon
                      color="ui.success"
                      name="checkCircle"
                      size={24}
                      title={null}
                    >
                      <WithTheme(Component)
                        className="emotion-3 emotion-0"
                        color="ui.success"
                        name="checkCircle"
                        size={24}
                        title={null}
                      >
                        <Component
                          className="emotion-3 emotion-0"
                          color="ui.success"
                          name="checkCircle"
                          size={24}
                          title={null}
                        >
                          <StyledSvg
                            className="emotion-3 emotion-0"
                            color="ui.success"
                            fill="currentcolor"
                            height={24}
                            title="checkCircle"
                            viewBox="0 0 24 24"
                            width={24}
                          >
                            <svg
                              className="emotion-0 emotion-1 emotion-2"
                              color="ui.success"
                              fill="currentcolor"
                              height={24}
                              title="checkCircle"
                              viewBox="0 0 24 24"
                              width={24}
                            >
                              <path
                                d={null}
                              />
                            </svg>
                          </StyledSvg>
                        </Component>
                      </WithTheme(Component)>
                    </Icon>
                  </div>
                </Box>
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-11 emotion-12"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Box>
      </div>
    </Box>
  </Component>
</Alert.success>
`;

exports[`<Alert /> contained renders correctly 1`] = `
.emotion-6 {
  margin-bottom: 16px;
  background-color: greys.porcelain;
}

.emotion-2 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  margin-bottom: 0;
  text-align: left;
}

.emotion-4 {
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  max-width: default;
}

<Alert
  contained={true}
  mb={3}
  variant="default"
>
  <Component
    contained={true}
    mb={3}
    variant="default"
  >
    <Box
      bg="greys.porcelain"
      mb={3}
    >
      <div
        className="emotion-6 emotion-7"
      >
        <Container
          gutter="default"
          maxWidth="default"
        >
          <div
            className="emotion-4 emotion-5"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-2 emotion-3"
              >
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-0 emotion-1"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Container>
      </div>
    </Box>
  </Component>
</Alert>
`;

exports[`<Alert /> renders correctly 1`] = `
.emotion-6 {
  margin-bottom: 16px;
  background-color: greys.porcelain;
}

.emotion-4 {
  padding-left: 32px;
  padding-right: 32px;
}

.emotion-2 {
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  margin-bottom: 0;
  text-align: left;
}

<Alert
  mb={3}
  variant="default"
>
  <Component
    mb={3}
    variant="default"
  >
    <Box
      bg="greys.porcelain"
      mb={3}
    >
      <div
        className="emotion-6 emotion-5"
      >
        <Box
          px={4}
        >
          <div
            className="emotion-4 emotion-5"
          >
            <Flex
              py={4}
            >
              <div
                className="emotion-2 emotion-3"
              >
                <Text
                  hidden={false}
                  mb={0}
                  textAlign="left"
                  textStyle="text"
                >
                  <span
                    className="emotion-0 emotion-1"
                    hidden={false}
                  >
                    Hello world
                  </span>
                </Text>
              </div>
            </Flex>
          </div>
        </Box>
      </div>
    </Box>
  </Component>
</Alert>
`;
